release/k3s/k3s.go

```diff
@@ -0,0 +1,568 @@
+package k3s
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/google/go-github/v39/github"
+	"github.com/rancher/ecm-distro-tools/repository"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	ssh2 "golang.org/x/crypto/ssh"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	k8sUpstreamURL     = "https://github.com/kubernetes/kubernetes"
+	rancherRemote      = "k3s-io"
+	k8sRancherURL      = "git@github.com:k3s-io/kubernetes.git"
+	k8sUserURL         = "git@github.com:user/kubernetes.git"
+	k3sUpstreamRepoURL = "https://github.com/k3s-io/k3s"
+	gitconfig          = `
+	[safe]
+	directory = /home/go/src/kubernetes
+	[user]
+	email = %email%
+	name = %user%
+	`
+	dockerDevImage = `
+	FROM %goimage%
+	RUN apk add --no-cache bash git make tar gzip curl git coreutils rsync alpine-sdk
+	`
+)
+
+type Release struct {
+	OldK8SVersion string `json:"old_k8s_version"`
+	NewK8SVersion string `json:"new_k8s_version"`
+	OldK8SClient  string `json:"old_k8s_client"`
+	NewK8SClient  string `json:"new_k8s_client"`
+	OldK3SVersion string `json:"old_k3s_version"`
+	NewK3SVersion string `json:"new_k3s_version"`
+	ReleaseBranch string `json:"release_branch"`
+	Workspace     string `json:"workspace"`
+	Handler       string `json:"handler"`
+	Email         string `json:"email"`
+	Token         string `json:"token"`
+}
+
+func NewReleaseFromConfig(c *cli.Context) (*Release, error) {
```

Comment:
Since this is a library function we should abstract away the CLI from it and pass in the dependency and keep knowledge of the CLI only to the utility itself.